### DISCUSSION TITLE
EAP question

### DISCUSSION THREAD

**Discussion by FrontWhite** [strongswan/strongswan#1917] [Q&A] [answered]
I am a beginner and have the following questions to ask you:
1. Check whether EAP authentication can be used only with X.509 certificates. Can it be used with PSK authentication？
2. Can EAP authentication be used in Site-to-Site mode or only in Remote Access  mode? (I only see Remote Access config example at https://docs.strongswan.org/docs/5.9/config/IKEv2.html)
3. If EAP authentication is used, what necessary preparations are required, such as what configurations are enabled?

Hope to get your answer, thank you very much!

**Comment by misch42** [comment]
First hit @goolge: https://www.intel.de/content/www/de/de/support/articles/000006999/wireless/legacy-intel-wireless-products.html

EAP offerst, amongst others, MD5 or CHAP methods for password auth.

Topology is not related to auth methods. So yes, you can auth site2site connecitons with PSK.

3: See doc.

**Comment by tobiasbrunner** [comment, marked as answer]
>  1. Check whether EAP authentication can be used only with X.509 certificates.

No, there are various EAP methods that use username/password. But note that not all implementations support all of them (most commonly supported e.g. by Windows or macOS/iOS is EAP-MSCHAPv2). And note that the server still has to be authenticated with a certificate first.

>  2. Can EAP authentication be used in Site-to-Site mode or only in Remote Access  mode? (I only see Remote Access config example at https://docs.strongswan.org/docs/5.9/config/IKEv2.html)

For site-to-site setups EAP doesn't provide any advantage over regular certificate or PSK authentication (except for the EAP-TNC use case maybe). And only a few EAP methods can even be used that way (requires mutual authentication that can be used in an EAP-only setup) and all of these require certificates (e.g. EAP-TLS). Generally, EAP is asymmetrical and mostly used in remote access setups (e.g. with a RADIUS server at the remote end).

> 3. If EAP authentication is used, what necessary preparations are required, such as what configurations are enabled?

Not sure exactly what you are asking, but refer to the docs for config options and examples.

**Reply by FrontWhite** [reply]
thanks
